CMakeLists.txt

```diff
@@ -540,3 +540,13 @@ string(REGEX REPLACE "/" "\\\\\\\\" NSIS_IMAGE_PATH ${NSIS_IMAGE_PATH})
 set(CPACK_NSIS_INSTALLER_MUI_ICON_CODE "!define MUI_WELCOMEFINISHPAGE_BITMAP \\\"${NSIS_IMAGE_PATH}sidebar.bmp\\\"\n!define MUI_HEADERIMAGE_BITMAP \\\"${NSIS_IMAGE_PATH}header.bmp\\\"\n!define MUI_ICON \\\"${NSIS_IMAGE_PATH}sfml.ico\\\"")
 
 include(CPack)
+
+if(NOT CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+    return()
+endif()
+
+add_custom_target(format
+    COMMAND git ls-files -z -- '*.hpp' '*.inl' '*.cpp' ':!:extlibs/*' | xargs -0 clang-format -i
+    # -z makes git separate lines with null, instead of \n
+    # -0 makes xargs separate by null, not \n, and makes other characters be taken literally
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
```

Comment:
We might need to quote the WORKING_DIRECTORY since someone could have a source directory that includes spaces.